src/output.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Contributors to the Parsec project.
```

Comment:
I took a look around for some libraries to handle logging/style but I couldn't find anything that was as flexible as `ansi_term`, which is what I've opted for here. Logging felt a little bit extreme, since we're not really 'logging' as much as we are just writing to std{out,err}. Happy to change this if you think it's a better long term solution though!

Reply:
No, this makes sense, it's not really logging in the end :) I think I was mostly bothered by the thought of having to set the colour for each line that we write, feels a bit extreme.